core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -116,6 +119,19 @@ private void initializeCatalogTables() throws InterruptedException, SQLException
       LOG.debug("Creating table {} to store iceberg catalog", JdbcUtil.CATALOG_TABLE_NAME);
       return conn.prepareStatement(JdbcUtil.CREATE_CATALOG_TABLE).execute();
     });
+
+    connections.run(conn -> {
+      DatabaseMetaData dbMeta = conn.getMetaData();
+      ResultSet tableExists = dbMeta.getTables(null, null, JdbcUtil.CREATE_NAMESPACE_PROPERTIES_TABLE, null);
+
+      if (tableExists.next()) {
+        return true;
```

Comment:
Yeah, I think the naming can't be be better. Tried to keep it simple and effective.

Not sure I follow the `getTables` question. It's checking if there exists a row in the Tables table where this namespace exists. 
```
protected static final String GET_NAMESPACE_SQL = "SELECT " + TABLE_NAMESPACE + " FROM " + CATALOG_TABLE_NAME +
      " WHERE " + CATALOG_NAME + " = ? AND " + TABLE_NAMESPACE + " LIKE ? LIMIT 1";
```
not sure what you mean about the CREATE TABLE DDL being used? Could you clarify?

Reply:
So what you referenced in your response to me seems correct.

But the code isn't using `GET_NAMESPACE_SQL` in that `getTables` call. It's using `CREATE_NAMESPACE_PROPERTIES_SQL`, which is copied at the bottom.

I think you just need to update this line https://github.com/apache/iceberg/pull/3275/files#diff-a5f7966c7493d463facbf7e76fe0fa3db1b637d29ee12e5b3628a5e5dd289e0aR125 to use `GET_NAMESPACE_SQL` like you mentioned.

```java
  protected static final String CREATE_NAMESPACE_PROPERTIES_TABLE =
      "CREATE TABLE " + NAMESPACE_PROPERTIES_TABLE_NAME +
          "(" +
          CATALOG_NAME + " VARCHAR(255) NOT NULL," +
          NAMESPACE_NAME + " VARCHAR(255) NOT NULL," +
          NAMESPACE_PROPERTY_KEY + " VARCHAR(5500)," +
          NAMESPACE_PROPERTY_VALUE + " VARCHAR(5500)," +
          "PRIMARY KEY (" + CATALOG_NAME + ", " + NAMESPACE_NAME + ", " + NAMESPACE_PROPERTY_KEY + ")" +
          ")";
```

Reply:
Oh, I think there's some confusion here. The right thing should be ```NAMESPACE_PROPERTIES_TABLE_NAME``` instead of the create statement. I fixed it locally. I'll push it in the later commit.
GET_NAMESPACE_SQL was right but for namespaceExists.